tests/routes/test_expressions.py

```diff
@@ -0,0 +1,43 @@
+from django.test import TestCase
+
+from conman.routes.expressions import CharCount
+from conman.routes.models import Route
+
+from .factories import RouteFactory
+
+
+class TestCharCount(TestCase):
+    """Tests for CharCount."""
+    def test_query(self):
+        """Match the exact value of the generated query."""
+        # The "only" here is handy to keep the query as short as possible.
+        qs = Route.objects.only('id').annotate(level=CharCount('url', char='/'))
+        # Excuse the line wrapping here -- I wasn't sure of a nice way to do it.
+        # I decided it was better to just keep it simple.
+        expected = (
+            'SELECT "routes_route"."id", ' +
+            'CHAR_LENGTH("routes_route"."url") - ' +
+            'CHAR_LENGTH(REPLACE("routes_route"."url", \'/\', \'\')) AS "level" ' +
+            'FROM "routes_route"'
+        )
+        self.assertEqual(str(qs.query), expected)
```

Comment:
`str(qs.query)` isn't perfect, since it does the interpolation of any params in python, not in sql. It might be worth using [`connection.queries`](https://docs.djangoproject.com/en/2.0/faq/models/#how-can-i-see-the-raw-sql-queries-django-is-running) instead.

Reply:
Thanks for the heads up -- I'll give that a shot.